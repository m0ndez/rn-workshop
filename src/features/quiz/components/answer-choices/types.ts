export type AnswerChoicesProps = {
  label?: string;
  choices: string[];
  onValueChange: (choice: string) => void;
  value: string;
};
