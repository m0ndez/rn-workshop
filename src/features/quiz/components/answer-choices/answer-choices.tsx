import { FC, useState } from "react";
import { FlatList, View } from "react-native";
import { RadioButton, Text } from "react-native-paper";
import { AnswerChoicesProps } from "./types";

export const AnswerChoices: FC<AnswerChoicesProps> = (props) => {
  const { choices, onValueChange, label, value } = props;

  return (
    <View>
      {label && <Text variant="labelLarge">{label}</Text>}
      <RadioButton.Group onValueChange={onValueChange} value={value}>
        <FlatList
          scrollEnabled={false}
          data={choices}
          keyExtractor={(item) => item}
          renderItem={({ item }) => (
            <RadioButton.Item
              labelVariant="bodyMedium"
              label={item}
              value={item}
              mode="android"
            />
          )}
        />
      </RadioButton.Group>
    </View>
  );
};
