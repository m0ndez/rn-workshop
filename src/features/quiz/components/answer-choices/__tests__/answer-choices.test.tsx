import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import { AnswerChoices } from "../answer-choices";

describe("AnswerChoices component", () => {
  const choices = ["Choice 1", "Choice 2", "Choice 3"];
  const onValueChange = jest.fn();

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <AnswerChoices
          choices={choices}
          onValueChange={onValueChange}
          value=""
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onValueChange when a choice is selected", () => {
    const { getByText } = render(
      <AnswerChoices choices={choices} onValueChange={onValueChange} value="" />
    );

    fireEvent.press(getByText("Choice 1"));
    expect(onValueChange).toHaveBeenCalledWith("Choice 1");

    fireEvent.press(getByText("Choice 2"));
    expect(onValueChange).toHaveBeenCalledWith("Choice 2");

    fireEvent.press(getByText("Choice 3"));
    expect(onValueChange).toHaveBeenCalledWith("Choice 3");
  });
});
