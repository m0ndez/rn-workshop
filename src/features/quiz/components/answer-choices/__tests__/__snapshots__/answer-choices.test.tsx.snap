// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerChoices component renders correctly 1`] = `
<View>
  <View
    accessibilityRole="radiogroup"
  >
    <RCTScrollView
      data={
        [
          "Choice 1",
          "Choice 2",
          "Choice 3",
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="Choice 1"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                undefined,
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        false,
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "textAlign": "left",
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
              >
                Choice 1
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(73, 69, 79, 1)",
                      "borderRadius": 10,
                      "borderWidth": 2,
                      "height": 20,
                      "margin": 8,
                      "width": 20,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="Choice 2"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                undefined,
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        false,
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "textAlign": "left",
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
              >
                Choice 2
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(73, 69, 79, 1)",
                      "borderRadius": 10,
                      "borderWidth": 2,
                      "height": 20,
                      "margin": 8,
                      "width": 20,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="Choice 3"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                undefined,
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        false,
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "textAlign": "left",
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
              >
                Choice 3
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(73, 69, 79, 1)",
                      "borderRadius": 10,
                      "borderWidth": 2,
                      "height": 20,
                      "margin": 8,
                      "width": 20,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
